modules/client/src/main/java/org/apache/ignite/internal/client/compute/ClientCompute.java

```diff
@@ -271,35 +269,15 @@ private CompletableFuture<PayloadInputChannel> executeOnOneNode(
         return ch.serviceAsync(
                 ClientOp.COMPUTE_EXECUTE,
                 w -> {
-                    if (w.clientChannel().protocolContext().clusterNode().name().equals(node.name())) {
-                        w.out().packNil();
-                    } else {
-                        w.out().packString(node.name());
-                    }
-
+                    packNodeNames(w.out(), nodes);
                     packJob(w.out(), units, jobClassName, options, args);
                 },
                 ch -> ch,
-                node.name(),
+                null,
```

Comment:
Don't pass null as `preferredNode`. Let's pass random node from `nodes` to increase the chance of local execution.

This is especially important when `nodes` has only one node and a direct connection exists - guaranteed single network call.